java/flight/flight-core/src/main/java/org/apache/arrow/flight/grpc/AddWritableBuffer.java

```diff
@@ -88,11 +87,8 @@
    * @param buf The buffer to add.
    * @param stream The Candidate OutputStream to add to.
    * @return True if added. False if not possible.
-   * @throws IOException on error
    */
-  public static boolean add(ByteBuf buf, OutputStream stream) throws IOException {
-    buf.readBytes(stream, buf.readableBytes());
```

Comment:
while I'm trying to parse the code, its not clear to me why this didn't end up doing a double send? (it is markers in the buf already indicate it is drained?

Reply:
Yes - `readBytes` increments the reader index and so the subsequent getBytes call wouldn't do anything.

https://github.com/apache/arrow/blob/6b85f6eba6d8403dbd549212fbc86f0e8ee192d0/java/flight/flight-core/src/main/java/org/apache/arrow/flight/ArrowMessage.java#L441